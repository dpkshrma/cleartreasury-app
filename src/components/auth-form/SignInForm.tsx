import React, { FunctionComponent } from "react";
import Link from "next/link";
import { Input, Button } from "@clear-treasury/design-system";
import { FieldErrors, useFormContext } from "react-hook-form";
import { FormType } from "./types";

export interface SignInFormProps {
  errors?: FieldErrors;
  loading?: boolean;
  formType?: FormType;
}

export const SignInForm: FunctionComponent<SignInFormProps> = ({
  errors,
  loading,
  formType,
}) => {
  if (formType !== FormType.signInForm) return null;

  const { register } = useFormContext();
  return (
    <>
      <Input
        type="text"
        label="Email address"
        placeholder="Enter your email"
        errors={errors}
        {...register("email")}
      />

      <Input
        type="password"
        label="Password"
        placeholder="Enter your password"
        errors={errors}
        {...register("password")}
      />

      <Link href="/reset-password">
        <a className="text-green-700 text-sm mb-16 cursor-pointer">
          Forgot your password?
        </a>
      </Link>

      <Button size={Button.Size.LARGE} loading={loading}>
        Sign in
      </Button>
    </>
  );
};
