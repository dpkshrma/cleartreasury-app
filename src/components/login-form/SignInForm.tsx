import React, { Ref } from "react";
import Link from "next/link";
import { Input, Button } from "@clear-treasury/design-system";

export interface SignInFormProps {
  passwordRef: Ref<HTMLInputElement | null>;
  errors: any;
  loading: boolean;
}

export const SignInForm: React.FunctionComponent<SignInFormProps> = ({
  passwordRef,
  errors,
  loading,
}) => (
  <>
    <Input
      type="password"
      name="password"
      label="Password"
      placeholder="Enter your password"
      ref={passwordRef}
      errors={errors}
    />

    <Link href="/reset-password">
      <a className="text-green-700 text-sm mb-16 cursor-pointer">
        Forgot your password?
      </a>
    </Link>

    <Button size={Button.Size.LARGE} loading={loading}>
      Sign in
    </Button>
  </>
);
