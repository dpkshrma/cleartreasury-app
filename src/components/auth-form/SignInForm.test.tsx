import * as React from "react";
import { render } from "../../../test/testUtils";
import { Default } from "./SignInForm.stories";

describe("SignInForm", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Default />);
    expect(asFragment()).toMatchSnapshot();
  });
});
