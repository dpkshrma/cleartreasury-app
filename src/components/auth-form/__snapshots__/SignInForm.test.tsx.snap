// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex justify-center flex-col space-y-6"
  >
    <h1
      class="block w-full text-center text-gray-800 text-2xl"
    >
      Sign in to your account
    </h1>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="email"
        id="email-label"
      >
        Email address
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="password"
        id="password-label"
      >
        Password
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="password"
        name="password"
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <a
      class="text-green-700 text-sm mb-16 cursor-pointer"
      href="/reset-password"
    >
      Forgot your password?
    </a>
    <button
      class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
    >
      Sign in
    </button>
  </form>
</DocumentFragment>
`;
